music_assistant/providers/radioparadise/__init__.py

```diff
@@ -0,0 +1,369 @@
+    "aac-128": {"content_type": ContentType.AAC, "sample_rate": 44100, "bit_depth": 16},
+    "aac-64": {"content_type": ContentType.AAC, "sample_rate": 44100, "bit_depth": 16},
+}
+
+# Ordered list of formats for fallback logic
+FALLBACK_ORDER = ["flac", "aac-320", "mp3-192", "aac-128", "aac-64"]
+
+
+async def setup(
+    mass: MusicAssistant, manifest: ProviderManifest, config: ProviderConfig
+) -> ProviderInstanceType:
+    """Initialize provider(instance) with given configuration."""
+    return RadioParadiseProvider(mass, manifest, config)
+
+
+async def get_config_entries(
+    mass: MusicAssistant,  # noqa: ARG001
+    instance_id: str | None = None,  # noqa: ARG001
+    action: str | None = None,  # noqa: ARG001
+    values: dict[str, ConfigValueType] | None = None,  # noqa: ARG001
+) -> tuple[ConfigEntry, ...]:
+    """Return Config entries to setup this provider."""
+    return (
+        ConfigEntry(
+            key="stream_format",
+            type=ConfigEntryType.STRING,
+            label="Stream Quality",
+            description="Choose the audio quality/format for streams",
+            required=True,
+            default_value="flac",
+            options=[
+                ConfigValueOption(title="FLAC (Lossless)", value="flac"),
+                ConfigValueOption(title="AAC 320kbps", value="aac-320"),
+                ConfigValueOption(title="MP3 192kbps", value="mp3-192"),
+                ConfigValueOption(title="AAC 128kbps", value="aac-128"),
+                ConfigValueOption(title="AAC 64kbps", value="aac-64"),
+            ],
+        ),
+    )
+
+
+class RadioParadiseProvider(MusicProvider):
+    """Radio Paradise Music Provider for Music Assistant."""
+
+    def __init__(self, mass: MusicAssistant, manifest: ProviderManifest, config: ProviderConfig):
+        """Initialize the provider."""
+        super().__init__(mass, manifest, config)
+        self._channels_cache: dict[str, dict[str, Any]] = RADIO_PARADISE_CHANNELS.copy()
+        self._stream_format: str = cast("str", self.config.get_value("stream_format", "flac"))
+        self._current_stream_details: StreamDetails | None = None
+        self._monitor_task: asyncio.Task[None] | None = None
+
+    @property
+    def supported_features(self) -> set[ProviderFeature]:
+        """Return the features supported by this Provider."""
+        return {
+            ProviderFeature.BROWSE,
+            ProviderFeature.LIBRARY_RADIOS,
+            ProviderFeature.TRACK_METADATA,
+        }
+
+    @property
+    def is_streaming_provider(self) -> bool:
+        """Return True if the provider is a streaming provider."""
+        return True
+
+    async def loaded_in_mass(self) -> None:
+        """Call after the provider has been loaded."""
+        if self.mass and hasattr(self.mass, "music") and hasattr(self.mass.music, "sync_library"):
+            self.mass.create_task(self.mass.music.sync_library(self.instance_id, MediaType.RADIO))
```

Comment:
this needs to be removed completely, just leave out this method

Reply:
Done